src/docker-tags.ts

```diff
@@ -2,3 +2,16 @@ import dockerContainerConfig from '../docker/containers.json'
 
 export const UPDATER_IMAGE_NAME = dockerContainerConfig.updater
 export const PROXY_IMAGE_NAME = dockerContainerConfig.proxy
+
+const imageNamePattern =
+  '^(?<repository>(([a-zA-Z0-9._-]+([:[0-9]+[^/]))?([a-zA-Z0-9._/-]+)?))((:[a-zA-Z0-9._/-]+)|(@sha256:[a-zA-Z0-9]{64}))?$'
```

Comment:
CRF https://regex101.com/r/CzIkfY/1

This is a port of a Python regex, it likely isn't exhaustive but it covers the test cases above so I think it's good enough no to cause any obvious pitfalls later